packages/dappmanager/src/calls/ethicalMetrics.ts

```diff
@@ -78,6 +79,14 @@ export async function enableEthicalMetrics({
       if (!container.running)
         await dockerContainerStart(container.containerName);
 
+    //Checking if Ethical Metrics API is up before registering
+    let isEthicalUp = await isApiUp();
+    while (!isEthicalUp) {
```

Comment:
this might cause a potential infinite loop issue because setInterval is non-blocking and schedules an asynchronous operation, the loop does not wait for isApiUp() to complete before continuing. This means the while loop could potentially create an enormous number of intervals in a very short time

Reply:
avoid using while,